test/helpers/block.ts

```diff
@@ -193,15 +193,32 @@ export const verifyBlockFees = async (
                   gasFee = ethTxWrapper.asEip1559.maxFeePerGas.toBigInt();
                 }
 
+                const hash = events
+                  .find((event) => event.section === "ethereum" && event.method === "Executed")!
+                  .data[2].toHex();
+
+                const receipt = await context.viem("public").getTransactionReceipt({ hash });
+
                 let effectiveTipPerGas = gasFee - baseFeePerGas;
                 if (effectiveTipPerGas > priorityFee) {
                   effectiveTipPerGas = priorityFee;
                 }
+                const effectiveGas = receipt!.gasUsed;
```

Comment:
This breaks some assumptions above, the `gasUsed` obtained above should already be the effective gas, since we converted to `ref_time`.

Reply:
What do you mean by "gasUsed obtained above"

Reply:
There is now an unused `gasUsed` variable in the lines above

Reply:
I can remove it. since the effective gas can be either ref_time based, pov based, or storage based

```
let effective_gas = core::cmp::max(core::cmp::max(used_gas, pov_gas), storage_gas);
```
https://github.com/moonbeam-foundation/frontier/blob/moonbeam-polkadot-stable2409/frame/evm/src/runner/stack.rs#L363

Reply:
`used_gas` is not the same as `ref_time`, it is the standard gas calculated in evm

Reply:
reverted